src/server/task.rs

```diff
@@ -214,8 +215,12 @@ impl Task {
             })
             .collect();
 
+        let instance_id = self.instance_id;
+        self.instance_id += 1;
+
         ToWorkerMessage::ComputeTask(ComputeTaskMsg {
             id: self.id,
+            instance_id,
```

Comment:
This could be combined with Id into a single type, since both seem to be a part of the effective Task ID.

Reply:
Maybe, but instance id exists only for workers (the server only work with task_ids until is is not dispatched), in a single worker there is always only one instance id per task, so it is ok to just use task_id as a unique identifier both internally and with communication with server